### PR TITLE
Update documentation on how to authenticate the client

### PR DESCRIPTION
**Pull request by erlendvollset**

### CODE REVIEW

File: docs/source/cognite.rst [hunk 1]
```diff
@@ -2,13 +2,28 @@ Quickstart
 ==========
 Authenticate
 ------------
-The preferred way to authenticate against the Cognite API is by setting the :code:`COGNITE_API_KEY` environment variable. All examples in this documentation require that the variable has been set.
+
+The preferred way to authenticating against the Cognite API is using native tokens. To enable this, the CogniteClient
```
Comment (muradsater) [inline docs/source/cognite.rst:6]: native tokens is only an internal Cognite thing, it's should be named OpenID Connect (OIDC)

File: docs/source/cognite.rst [hunk 2]
```diff
@@ -2,13 +2,28 @@ Quickstart
 ==========
 Authenticate
 ------------
-The preferred way to authenticate against the Cognite API is by setting the :code:`COGNITE_API_KEY` environment variable. All examples in this documentation require that the variable has been set.
+
```
Comment (muradsater) [inline docs/source/cognite.rst:5]: A CDF project can be setup to either authenticate with OpenID Connect (OIDC) or API keys. OIDC is preferred and all new project will be setup with that. Projects setup before May 2021 will most likely use API keys unless it has been migrated to OIDC.   (it's not something the user can decide himself)

File: docs/source/cognite.rst (hunk 1 above)
Comment (erlendvollset) [inline docs/source/cognite.rst:6]:
```suggestion
The preferred way to authenticating against the Cognite API is using OpenID Connect (OIDC). To enable this, the CogniteClient
```

File: docs/source/cognite.rst [hunk 3]
```diff
@@ -2,13 +2,28 @@ Quickstart
...
+    >>> from cognite.client import CogniteClient
+    >>> def token_provider():
+    >>>     ...
+    >>> c = CogniteClient(token=token_provider)
+
+For details on different ways of implementing the token provider, take a look at
+`this guide <https://github.com/cognitedata/python-oidc-authentication>`_.
+
+In some cases you may also want to authenticate using an API key. You can do this by setting the following environment
```
Comment (erlendvollset) [inline docs/source/cognite.rst:19]:
```suggestion
If OIDC has not been enabled for your CDF project, you will want to authenticate using an API key. You can do this by setting the following environment
```

**Review by muradsater — APPROVED** [review]

**Review by me-ydv-5 — APPROVED** [review]

**Pull request merged**